CSharpSourceCode/HarmonyPatches/CampaignPatches.cs

```diff
@@ -124,14 +175,14 @@ public static void Postfix(ref Settlement __result, Kingdom __instance)
         //This behaviour contains hardcoded hero / lord references that are not present because we skip loading the vanilla files.
         [HarmonyPrefix]
         [HarmonyPatch(typeof(BackstoryCampaignBehavior), "RegisterEvents")]
-        public static bool Prefix3()
+        public static bool WhyIsThisNeeded()
```

Comment:
:D. Still would be curious about the story of it

Reply:
If we don't need the method, it should be removed

Reply:
As it says right there in the comment. BackstoryCampaignBehavior is a vanilla campaignbehaviour that gets loaded on campaign load from vanilla code. It contains a hardcoded string reference to a vanilla NPC, which we are not loading, so it cannot find an expected object and the game crashes. Ideally we should be able to completely remove this BackstoryCampaignBehavior as it is not even really needed by us (it contains vanilla story related stuff) which we do in Submodule.cs on line 109:
starter.CampaignBehaviors.RemoveAllOfType(typeof(BackstoryCampaignBehavior));
yet the onNewGame event still fires inside BackstoryCampaignBehavior where the crash happens despite this. So i ended having this workaround. If anyone can suggest a better way, pls do.